security/pkg/stsservice/server/server.go

```diff
@@ -109,6 +109,21 @@ func (s *Server) ServeStsRequests(w http.ResponseWriter, req *http.Request) {
 		s.sendErrorResponse(w, invalidRequest, validationError)
 		return
 	}
+	// Authentication using the original JWT - Google requests use a Scope.
+	// This is used for authenticating directly with the token, for XDS/Citadel
+	if reqParam.Scope == "" {
+		stsServerLog.Infoa("STS local request: ", reqParam.Audience)
+		tokInfo := &stsservice.StsResponseParameters{
+			AccessToken: reqParam.SubjectToken,
+			TokenType:   "Bearer",
+		}
+		statusJSON, err := json.MarshalIndent(tokInfo, "", " ")
+		if err != nil {
+			stsServerLog.Warna("Failed to marshal", err)
+		}
+		s.sendSuccessfulResponse(w, statusJSON)
```

Comment:
So you basically want to leverage the validateStsRequest() at line 104, and send back this jwt if validation passes?

Reply:
Not really - I just want to parse the response and send back the token. Validation is not very relevant.